lab08-kaylee/lib/storage.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const debug = require('debug')('http:storage');
+const storage = {};
+
+module.exports = exports = {};
+
+exports.createNote = function(schema, note) {
+  debug('#createNote');
+  return new Promise((resolve, reject) => {
+    if(!schema) return reject(new Error('schema required'));
+    if(!note) return reject(new Error('note required'));
+    if(!storage[schema]) storage[schema] = {};
+
+    storage[schema][note.id] = note;
+    resolve(note);
+  });
+};
+
+exports.fetchNote = function(schema, id) {
+  debug('#fetchNote');
+
+  return new Promise((resolve, reject) => {
+    if(!schema) return reject(new Error('schema required'));
+    if(!id) return reject(new Error('id required'));
+
+    let schemaName = storage[schema];
+    if(!schemaName) return reject(new Error('schema not found'));
+
+    let note = schemaName[id];
+    if(!note) return reject(new Error('note not found'));
+    resolve(note);
+  });
+};
+
+exports.updateNote = function(schema, id, newNote) {
```

Comment:
glad you put updateNote as it's own bit of functionality. it's easier to unit test, plus it isn't reliant on other routes working! well done.